### PR TITLE
Remove separate os lib module + fixes + readme

### PR DESCRIPTION
**Pull request by fwbrasil**
The code was in a separate module to isolate the dependency on `os-lib` but, given that the dependency is not needed anymore, I think keeping it in `kyo-core` is ideal so users don't need to import the separate module.

I've also made a few refactorings to avoid polluting the `kyo` namespace and added a misssing IOs suspension.

### CODE REVIEW

File: kyo-core/jvm/src/main/scala/kyo/Process.scala [hunk 1]
```diff
@@ -0,0 +1,235 @@
+package kyo
```
Comment (fwbrasil) [inline kyo-core/jvm/src/main/scala/kyo/Process.scala:1]: I tried to keep the file rename in the git history but I couldn't make it work :(

File: kyo-core/jvm/src/main/scala/kyo/files.scala [hunk 2]
```diff
@@ -127,7 +127,7 @@ class Files(val path: List[String]):
       */
     def readLinesStream(charset: Charset = java.nio.charset.StandardCharsets.UTF_8)(using Trace): Stream[Unit, String, Fibers] < Resources =
         readLoop[String, String, BufferedReader](
-            JFiles.newBufferedReader(toJava, Charset.defaultCharset()),
+            IOs(JFiles.newBufferedReader(toJava, Charset.defaultCharset())),
```
Comment (fwbrasil) [inline kyo-core/jvm/src/main/scala/kyo/files.scala:130]: Fix to include a missing `IOs` suspension.

**Comment by fwbrasil** [comment]
cc/ @pablf

File: kyo-core/jvm/src/main/scala/kyo/Process.scala (hunk 1 above)
Comment (fwbrasil) [reply kyo-core/jvm/src/main/scala/kyo/Process.scala:1]: It seems it actually worked but github doesn't show it: https://github.com/getkyo/kyo/pull/471/commits/d2b781bca6d72aca0d3276c8a0fd2b6c7b35007b

**Review by hearnadam — APPROVED** [review]

**Comment by fwbrasil** [comment]
> @fwbrasil LGTM, it seems better organized this way! Maybe it's more coherent with the rest of the library to use Files.scala instead of files.scala

Yeah, our naming patterns are a mess at the moment. I'll mention reviewing the file names as well in https://github.com/getkyo/kyo/issues/317.

**Pull request merged**